runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -80,11 +80,13 @@ static TR::SymbolReferenceTable::CommonNonhelperSymbol equivalentAtomicIntrinsic
    switch (rm)
       {
       case TR::sun_misc_Unsafe_getAndSetInt:
-      case TR::sun_misc_Unsafe_getAndSetLong:
            return TR::SymbolReferenceTable::atomicSwapSymbol;
+      case TR::sun_misc_Unsafe_getAndSetLong:
+           return TR::Compiler->target.is64Bit() ? TR::SymbolReferenceTable::atomicSwapSymbol : TR::SymbolReferenceTable::lastCommonNonhelperSymbol;
       case TR::sun_misc_Unsafe_getAndAddInt:
-      case TR::sun_misc_Unsafe_getAndAddLong:
            return TR::SymbolReferenceTable::atomicFetchAndAddSymbol;
+      case TR::sun_misc_Unsafe_getAndAddLong:
+           return TR::Compiler->target.is64Bit() ? TR::SymbolReferenceTable::atomicFetchAndAddSymbol : TR::SymbolReferenceTable::lastCommonNonhelperSymbol;
```

Comment:
IMO multiple return points make it hard see the exit points of functions where there is a default "do nothing" return value. Perhaps this may be more readible:

```
static TR::SymbolReferenceTable::CommonNonhelperSymbol equivalentAtomicIntrinsic(TR::RecognizedMethod rm)
   {
   auto result = TR::SymbolReferenceTable::lastCommonNonhelperSymbol;

   switch (rm)
      {
      case TR::sun_misc_Unsafe_getAndSetInt:
         result = TR::SymbolReferenceTable::atomicSwapSymbol;
         break;
      case TR::sun_misc_Unsafe_getAndSetLong:
         if (TR::Compiler->target.is64Bit())
            {
            result = TR::SymbolReferenceTable::atomicSwapSymbol;
            }
         break;
      case TR::sun_misc_Unsafe_getAndAddInt:
         result =  TR::SymbolReferenceTable::atomicFetchAndAddSymbol;
         break;
      case TR::sun_misc_Unsafe_getAndAddLong:
         if (TR::Compiler->target.is64Bit())
            {
            result = TR::SymbolReferenceTable::atomicFetchAndAddSymbol;
            }
        break;
      default:
         break;
      }

   return result;
}
```

Reply:
that is one style - some of the JIT recognition paths would be more complicated with a single return style so a multi return can be more clear, but making the 64bit condition clear is the main request I have rather than using an easily missed inline ternary.